ACE/bin/PerlACE/Process_Unix.pm

```diff
@@ -573,6 +573,193 @@ sub Spawn ()
+    }
+
+    my $pid_idx = index ($pattern, "%p");
+    if ($pid_idx != -1) {
+        substr ($pattern, $pid_idx, 2) = $self->{PROCESS};
+    }
+    elsif ($uses_pid != 0) {
+        $pattern = $pattern . "." . $self->{PROCESS};
+    }
+
+    my $timestamp_idx = index ($pattern, "%t");
+    my $core_file_path;
+    if ($timestamp_idx != -1) {
+        my $prefix = substr ($pattern, 0, $timestamp_idx);
+        my $suffix_len = length ($pattern) - $timestamp_idx - 2;
+        my $suffix = substr ($pattern, $timestamp_idx + 2, $suffix_len);
+
+        # Get the core file with latest timestamp.
+        my $dh;
+        if (!opendir ($dh, $path)) {
+            print STDERR "WARNING: print_stacktrace_linux: Couldn't opendir $path: $!\n";
+            return;
+        }
+        my @files = grep (/$prefix[0-9]+$suffix/, readdir ($dh));
+        my $latest_timestamp;
+        my $chosen_core_file;
+        foreach my $file (@files) {
+            my $timestamp_len = length ($file) - $timestamp_idx - $suffix_len;
+            my $timestamp = substr ($file, $timestamp_idx, $timestamp_len);
+            if (!defined $latest_timestamp) {
+                $latest_timestamp = $timestamp;
+                $chosen_core_file = $file;
+            }
+            elsif ($latest_timestamp < $timestamp) {
+                $latest_timestamp = $timestamp;
+                $chosen_core_file = $file;
+            }
+        }
+        closedir ($dh);
+        if (defined $chosen_core_file) {
+            $core_file_path = $path . "/" . $chosen_core_file;
+        }
+        else {
+            print STDERR "WARNING: print_stacktrace_linux: Could not determine a core file with timestamp\n";
+            return;
+        }
+    }
+    else {
+        $core_file_path = $path . "/" . $pattern;
+    }
+
+    my $debugger = "gdb";
+    if ($ENV{ACE_TEST_DEBUGGER}) {
+        $debugger = $ENV{ACE_TEST_DEBUGGER};
+    }
+    $self->print_stacktrace_common($exec_path, $core_file_path, $debugger);
+}
+
+sub print_stacktrace_darwin
+{
+    my $self = shift;
+    my $core_file_path = "/cores/core." . $self->{PROCESS};
+
+    my $debugger = "lldb";
+    if ($ENV{ACE_TEST_DEBUGGER}) {
+        $debugger = $ENV{ACE_TEST_DEBUGGER};
+    }
+    $self->print_stacktrace_common($self->Executable (), $core_file_path, $debugger);
+}
+
+sub print_stacktrace_common
+{
+    my $self = shift;
+    my $exec_path = shift;
+    my $core_file_path = shift;
+    my $preferred_db = shift;
+
+    if (!(-e $core_file_path)) {
+        print STDERR "WARNING: print_stacktrace_common: Core file $core_file_path does not exist\n";
+        return;
+    }
+    if (!defined $preferred_db) {
+        $preferred_db = "gdb";
+    }
+    my $preferred_cmd;
+    my $secondary_db;
+    my $secondary_cmd;
+    my $gdb_cmd = "gdb $exec_path -c $core_file_path -ex bt -ex quit";
+    my $lldb_cmd = "lldb $exec_path -c $core_file_path -o bt -o quit";
+
+    if ($preferred_db eq "gdb") {
```

Comment:
I think this will fail if a specific path is used in the environment variable or like I mentioned, my lldb is `lldb-12`. Maybe it needs to check that using something like `$preferred_db =~ /gdb/` or a separate environment like `ACE_TEST_DEBUGGER_KIND`.

Reply:
You're right, it'll fail with a path. I think using regex as you mentioned is good enough for those cases.